src/PaymentChannel.js

```diff
@@ -0,0 +1,114 @@
+import BigNumber from 'bignumber.js';
+
+import paymentChannelStateMessages from './payment_channel_state_service_pb';
+
+export default class PaymentChannel {
+  constructor(channelId, web3, account, paymentChannelStateServiceClient, mpeContract) {
+    this._channelId = channelId;
+    this._web3 = web3;
+    this._account = account;
+    this._mpeContract = mpeContract;
+    this._paymentChannelStateServiceClient = paymentChannelStateServiceClient;
+    this._state = {
+      nonce: new BigNumber(0),
+      lastSignedAmount: new BigNumber(0),
+    };
+  }
+
+  get channelId() {
+    return this._channelId;
+  }
+
+  get nonce() {
+    return this.state.nonce;
+  }
+
+  get lastSignedAmount() {
```

Comment:
Is it used anywhere?

Reply:
Am I right that it can be replaced by `state().lastSignedAmount` if so I would suggest removing this method to simplify API.

Reply:
Just convenience methods. We could remove `nonce` and `lastSignedAmount`.

Reply:
I would remove them in order to not write same code twice and make API cleaner.

Reply:
Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/1/commits/78cf9ae1c7ce7ce7b1812ea3d016dc3691663314)